docs/src/glossary.md

```diff
@@ -0,0 +1,85 @@
+# ChainRules Glossary
+
+This glossary serves as a quick reference for common terms used in the field of Automatic Differentiation, as well as those used throughout the documentation relating specifically to ChainRules.
+
+##Definitions:
+
+###Adjoint:
+
+The conjugate transpose of the Jacobian for a given function `f`.
+
+###Automatic Differentiation:
+
+Automatic Differentiation is the process of applying numerical methods to solving derivative problems, most often algorithmically, using computer programs to achieve high degrees of accuracy.
+
+###Derivative:
+
+The derivative of a function `y = f(x)` with respect to the independent variable `x` denoted `f'(x)` or `dy/dx` is the rate of change of the dependent variable `y` with respect to the change of the independent variable `x`. In multiple dimensions, we may refer to the gradient of a function.
+
+###Differential:
+
+The differential of a given function `y = f(x)` denoted `dy` is the product of the derivative function `f'(x)` and the increment of the independent variable `dx`. In multiple dimensions, it is the sum of these products across each dimension (using the partial derivative and the given independent variable's increment).
+
+In ChainRules, differentials are types ("differential types") and correspond to primal types. A differential should represent a difference between two primal values.
+
+####Natural Differential:
+
+A natural differential type for a given primal type is the type people would intuitively associate with representing the difference between two values of the primal type.
+
+####Structural Differential:
+
+If a given primal type `P` does not have a natural differential, we need to come up with one that makes sense. These are called structural differentials and are represented as `Composite{P, <:NamedTuple}`.
+
+####Semi-Structural Differential:
+
+A structural differential that contains at least one natural differential field.
+
+####Thunk:
+
```

Comment:
Thunk is actually more generally a term from computing.

Probably include a standard definition:
https://en.wikipedia.org/wiki/Thunk

In general  a way to represent thunks in julia is as anonoymous functions (including closures) which take no inputs.
e.g. `()->foo(x,y,z)`
And in ChainRules the specific `AbstractThunk` subtype is such a function that is setup to act like a differential.

Reply:
Added